File: Medium/Python/189. Rotate Array.py
class Solution(object):
    def rotate(self, nums, k):
        # In the event k is larger than len(nums),
        # simplify so to not rotate in circles
        n = len(nums)
        k %= n
        temp = [0] * n

        for i in range(n):
            temp[(i + k) % n] = nums[i]
        
        for i in range(n):
            nums[i] = temp[i]

File: Medium/Java/189. Rotate Array.java
class Solution {
    public void rotate(int[] nums, int k) {
        // In the event k is larger than nums.length,
        // simplify so to not rotate in circles
        int n = nums.length;
        k %= n;
        int[] temp = new int[n];
        
        for(int i = 0; i < n; i++){
            temp[(i + k) % n] = nums[i];
        }

        for(int i = 0; i < n; i++){
            nums[i] = temp[i];
        }
    }
}
